### PR TITLE
Refactor some quo2 components to use best practices

### PR DESCRIPTION
**Pull request by J-Son89**
Updated a few quo2 components to use `view` as their exported name,
to use style and view namespaces and to wrap some components in `with-theme` context HOC. 

To test - see pages using `tabs`, `dynamic-button` and `group tag`, `banner` and `step`

There should really be minimal changes here as it's all just internal refactoring to code formatting more than anything else

### CODE REVIEW

File: src/quo2/components/avatars/group_avatar/view.cljs [hunk 1]
```diff
@@ -13,20 +15,17 @@
 
 ;; TODO: this implementation does not support group display picture (can only display default group
 ;; icon).
-(defn group-avatar
+(defn view-internal
```
Comment (OmarBasem) [inline src/quo2/components/avatars/group_avatar/view.cljs]: missing private method declaration `defn-`

File: src/quo2/components/counter/step/view.cljs [hunk 2]
```diff
@@ -6,7 +6,7 @@
     [react-native.core :as rn]
     [utils.number]))
 
-(defn step-internal
+(defn view-internal
```
Comment (OmarBasem) [inline src/quo2/components/counter/step/view.cljs]: `defn-`

**Review by OmarBasem — APPROVED** [review]

File: src/quo2/components/buttons/dynamic_button/style.cljs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  {:margin-top    6
+   :margin-bottom 6
+   :margin-left   (case type
+                    :jump-to           0
+                    :mention           8
+                    :notification-down 2
+                    :notification-up   2
+                    :search-with-label 8
+                    :search            6
+                    :scroll-to-bottom  6)
```
Comment (ilmotta) [inline src/quo2/components/buttons/dynamic_button/style.cljs]: Better have a fallback to `nil` in `case`.

**Review by ilmotta — APPROVED** [review]

**Review by jo-mut — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @J-Son89 thank you for PR. Take a look a found issue:

## ISSUE 1: UI for 'Jump to' button is broken
### Steps:
1. Login
2. Check 'jump to' button

### Actual result:
 UI for 'Jump to' button is broken
image

### Expected result:
The collor for 'Jump to' corresponds to the selected color of the use

**Comment by VolodLytvynenko** [comment]
@J-Son89 I've observed that this issue persists in the latest nightly build. If the current scope of the PR refactor includes addressing this issue, kindly make the necessary fixes. If not, I'm prepared to create a separate issue. Thank you!

## ISSUE 2: [Andorid] The scroll button remains visible after reopening a chat
### Steps to Reproduce:
1. Open a chat.
2. Scroll up to appear the scroll button.
3. Close the chat.
4. Reopen the same chat.

### Actual Result:
Upon reopening the chat, the previously visible scroll button remains present even when the chat is opened in a manner that doesn't allow for scrolling downward.

https://www.dropbox.com/scl/fi/wvm8oyjya6y7is9z0fg6x/scroll.mp4?rlkey=kkihxrdf2pzvj4yghidif8zv7&dl=0


### Expected Result:
The scroll button should be hidden when the chat is reopened since the chat is displayed in a mode where scrolling down is not feasible.

### Device:
real device: Pixel 7a, Android 13

**Comment by J-Son89** [comment]
Hi @VolodLytvynenko - sure I can take a look at both of these. Will let you know!

**Comment by J-Son89** [comment]
Hey @VolodLytvynenko - I looked into issue 2 but this one is beyond my knowledge and better to have someone like @alwx, @flexsurfer 
 or perhaps @ibrkhalil address it. I have little knowledge of the composer/messenger so I have no idea how to fix it correctly. imo best to create a new issue for it :)

**Comment by J-Son89** [comment]
issue 1 I can not reproduce after rebasing 🤔

**Comment by VolodLytvynenko** [comment]
@J-Son89 Thank you for the clarification. I will recheck this PR one more time. Probably rebase helped with issue 1

**Comment by VolodLytvynenko** [comment]
Hi @J-Son89 indeed, issue 1 is resolved after rebase. One additional issue related to jump to is found:

## ISSUE 3: 'Jump To' Button Appearance Inconsistent Between Light and Dark Themes
### Steps to Reproduce:
1. Log in to the app.
2. Observe the appearance of the 'Jump To' button.

### Actual Result:
The 'Jump To' button is displayed with dark shades on the light theme and with a light appearance on the dark theme
image

### Expected Result:
The 'Jump To' button is displayed with light shades on the light theme and with a dark appearance on the dark theme

### ENVs:
- IOS
- Android

**Review by vkjr — APPROVED** [review]

**Comment by J-Son89** [comment]
apologies @VolodLytvynenko - should be addressed now 👍

**Review by ibrkhalil — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
@J-Son89 thank you for your work. PR is tested and ready to be merged

**Pull request merged**